packages/appeals-service-api/src/validators/appeals/appeals.validator.js

```diff
@@ -1,10 +1,21 @@
 const { insertAppeal } = require('./schemas/insert-appeal');
 const { updateAppeal } = require('./schemas/update-appeal');
+const { isAppealSubmitted } = require('../../services/appeal.service');
+const logger = require('../../lib/logger');
 const ApiError = require('../../error/apiError');
 
 const appealUpdateValidationRules = async (req, res, next) => {
   try {
     req.body = await updateAppeal.validate(req.body, { abortEarly: false });
+    logger.debug('Valid input format');
+
+    const appealId = req.body.id;
+    const pathId = req.params.id;
+
+    if (appealId && pathId !== appealId) {
+      next(ApiError.notSameId());
```

Comment:
I may be wrong here but I would perhaps call `return;` after this line, or `return next(ApiError.notSameId());`

Reason being, and this shouldn't cause a bug here, but I believe it could lead to some unexpected results...

What will happen here (again, afaik) is that the `next();` on line 19 will *also* be called, but at a much later point in the middleware stack. I'm guessing that would be unexpected / unwanted, as the way I read this, the flow is that this middleware is expected to end here (?) 

Whilst this should cause no bugs here today, it might be confusing in the future if logic was added after this `if` statement, as that logic would still get run but later in the chain. That might cause some really weird behaviour.

Of course, I may be entirely wrong. 

https://stackoverflow.com/a/43404213 explains it better than I have.